src/node_nanomsg.cc

```diff
@@ -205,23 +205,21 @@ NAN_METHOD(Shutdown) {
 
 
 NAN_METHOD(Send) {
-    NanScope();
-
-    int s = args[0]->Uint32Value();
-
-    if(!node::Buffer::HasInstance(args[1])) {
-        return NanThrowError("second Argument must be a Buffer.");
-    }
+  NanScope();
 
-    Local<Object> obj = args[1]->ToObject();
-    char* odata = node::Buffer::Data(obj);
-    size_t odata_len = node::Buffer::Length(obj);
-    int flags = args[2]->Uint32Value();
+  int flags = args[2].integer;
+  std::string *input;
 
-    // Invoke nanomsg function.
-    int ret = nn_send(s, odata, odata_len, flags);
+  if (node::Buffer::HasInstance(args[1]->ToObject())) {
+    v8::Handle<v8::Object> object = args[1]->ToObject();
```

Comment:
How expensive it ToObject()? Do we need to call it twice or can we move the creating of object to be outside the conditional?